command/variadic.go

```diff
@@ -0,0 +1,138 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/peak/s5cmd/objurl"
+	"github.com/peak/s5cmd/storage"
+)
+
+// Arg is a container type for supporting variadic arguments.
+type Arg struct {
+	origSrc *objurl.ObjectURL
+	obj     *storage.Object
+}
+
+// expandSources returns the full list of objects from the given src arguments.
+// If src is an expandable URL, such as directory, prefix or a glob, all
+// objects are returned by walking the source. It expands multiple resources asynchronously
+// and returns read-only arg channel.
+func expandSources(
+	ctx context.Context,
+	isRecursive bool,
+	dsturl *objurl.ObjectURL,
+	srcurls ...*objurl.ObjectURL,
+) (<-chan *Arg, error) {
+	if len(srcurls) == 0 {
+		return nil, fmt.Errorf("at least one source url is required")
+	}
+	// all sources share same client
+	client, err := storage.NewClient(srcurls[0])
```

Comment:
Injecting storage class to this function may help writing unit tests easily.

Reply:
Since we expect different results for each given source URL, only mocking the result of storage.List() will not work (since all sources share same client). We either need storage slice instead of source URL slice (then we can mock each one) or find a way to return results based on given URL. 
```

Reply:
@ilkinulas Could you please review expandSources function again. Since it only supports delete operations, I think it is much clear.

Reply:
I added unit tests.